src/drivers/distance_sensor/vl53l1x/vl53l1x.cpp

```diff
@@ -222,35 +241,56 @@ void VL53L1X::RunImpl()
 {
 	uint8_t dataReady = 0;
 
+	uint8_t roiCenter[] = {VL53L1X_ROI_FAR_RIGHT, VL53L1X_ROI_MID_RIGHT, VL53L1X_ROI_CENTER, VL53L1X_ROI_MID_LEFT, VL53L1X_ROI_FAR_LEFT};
+	static uint8_t zone = 0;
+
 	VL53L1X_CheckForDataReady(&dataReady);
 
 	if (dataReady == 1) {
 		collect();
 	}
 
 	ScheduleDelayed(VL53L1X_SAMPLE_RATE);
+
+	VL53L1X_SetROICenter(roiCenter[zone]);
+
+	// increment
+	zone++;
+
+	// reset counter check
+	if (zone > sizeof(roiCenter) - 1) {
+		zone = 0;
+	}
+
 }
 
 void VL53L1X::start()
 {
+
 	// Schedule the first cycle.
 	ScheduleNow();
+
 }
 
 int VL53L1X::init()
 {
 	int ret = PX4_OK;
-
+	uint8_t x, y;
```

Comment:
This isn't C89, you can declare these where you first use them below.

Reply:
done